src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataScriptTempFileStream.cs

```diff
@@ -0,0 +1,123 @@
+    {
+        private const short ScriptFileExpirationInDays = 30;
+
+        /// <summary>
+        /// This method writes the passed in scripts to a temporary file.
+        /// </summary>
+        /// <param name="serverName">The name of the server which will go on to become the name of the file.</param>
+        /// <param name="scripts">The generated scripts that will be written to the temporary file.</param>
+        public static void Write(string serverName, IEnumerable<string> scripts)
+        {
+            var tempFileName = $"{serverName}.tmp";
+            var generatedScripts = scripts.ToList();
+
+            try
+            {
+                var tempFilePath = Path.Combine(Path.GetTempPath(), tempFileName);
+                using (StreamWriter sw = new StreamWriter(tempFilePath, false))
+                {
+                    foreach (var script in generatedScripts)
+                    {
+                        sw.WriteLine(script);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.Warning($"Failed to write scripts to temporary file. Error: {ex.Message}");
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Reads the scripts associated with the provided server name.
+        /// </summary>
+        /// <param name="serverName">The name of the server to retrieve the scripts for.</param>
+        /// <returns>List containing all the scripts in the file.</returns>
+        public static IEnumerable<string> Read(string serverName)
+        {
+            var tempFileName = $"{serverName}.tmp";
+            var scripts = new List<string>();
+
+            try
+            {
+                var tempFilePath = Path.Combine(Path.GetTempPath(), tempFileName);
+                if (!File.Exists(tempFilePath))
+                {
+                    return scripts;
+                }
+
+                using (StreamReader sr = new StreamReader(tempFilePath))
+                {
+                    while (!sr.EndOfStream)
+                    {
+                        var line = sr.ReadLine();
+                        if (!String.IsNullOrWhiteSpace(line))
+                        {
+                            scripts.Add(line);
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.Warning($"Failed to read scripts from temporary file. Error: {ex.Message}");
+                throw;
+            }
+
+            return scripts;
+        }
+
+        /// <summary>
+        /// Determines if the script file for a server is too old.
+        /// </summary>
+        /// <param name="serverName">The name of the file associated with the given server name.</param>
+        /// <returns>True: The file was created within the expiration period; False: The script file needs to be created
```

Comment:
Isn't this reversed? True would imply that it needs to be updated according to the method name, but you're returning false for when that should occur.